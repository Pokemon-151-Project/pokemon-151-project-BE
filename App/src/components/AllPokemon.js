// This takes an array of Pokemon objects and returns an individual card (SinglePokemon.js) for each object
import React from "react";
import SinglePokemon from "./SinglePokemon";

const AllPokemon = (props) => {
  const { pokemonData, isShiny } = props;
  return (
    <main className="all-pokemon">
      {pokemonData.map((poke) => {
        return (
          <SinglePokemon
            poke={poke}
            key={poke.name}
            data-testid="single-pokemon"
            isShiny={isShiny}
          />
        );
      })}
    </main>
  );
};
export default AllPokemon;
