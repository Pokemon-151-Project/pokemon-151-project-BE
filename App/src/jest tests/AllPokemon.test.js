import AllPokemon from "../components/AllPokemon";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

//fake array with one pokemon for testing purposes
const fakePokemon = [
  {
    name: "Fake Pokemon Name",
    type: "Death Type",
    img: "https://google.com",
    dexEntry: "blah blah blah",
    height: "4m",
    weight: "50kg",
  },
];

//This is similar to the above array but with multiple pokemon for testing purposes
const multFakePokemon = [
  {
    name: "Fake Pokemon Name",
    type: "Death Type",
    img: "https://google.com",
    dexEntry: "blah blah blah",
    height: "4m",
    weight: "50kg",
  },
  {
    name: "Fake2",
    type: "Derp Type",
    img: "https://google.com/1234",
    dexEntry: "This pokemon does nothing",
    height: "5m",
    weight: "60kg",
  },
  {
    name: "Fake3",
    type: "Ground Type",
    img: "https://google.com/1234567",
    dexEntry: "This pokemon is god",
    height: "6m",
    weight: "80kg",
  },
];

test("[1] AllPokemon renders without errors", () => {
  render(<AllPokemon pokemonData={fakePokemon} />);
});

test("[2] AllPokemon renders correctly with certain props, then rerenders when those props change", () => {
  //renders first with empty props array, then renders with fakePokemon (defined above) to make sure it correctly renders with changed props.
  const { rerender } = render(<AllPokemon pokemonData={[]} />);

  const oldFakeName = screen.queryByText(/fake pokemon name/i);
  expect(oldFakeName).not.toBeInTheDocument();

  rerender(<AllPokemon pokemonData={fakePokemon} />);
  const newFakeName = screen.queryByText(/fake pokemon name/i);
  expect(newFakeName).toBeInTheDocument();
});

test("[3] Can render multiple pokemon", () => {
  //Passing in an array with multiple pokemon to make sure it renders all of them

  render(<AllPokemon pokemonData={multFakePokemon} />);

  const name1 = screen.queryByText(/fake pokemon name/i);
  const name2 = screen.queryByText(/fake2/i);
  const name3 = screen.queryByText(/fake3/i);
  const name4 = screen.queryByText(/fake4/i);
  expect(name1).toBeVisible();
  expect(name2).toBeVisible();
  expect(name3).toBeVisible();
  expect(name4).toBeFalsy();
});

test("[4] Matches snapshot from 5.8.22", () => {
  const component = renderer.create(
    <AllPokemon pokemonData={multFakePokemon} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
