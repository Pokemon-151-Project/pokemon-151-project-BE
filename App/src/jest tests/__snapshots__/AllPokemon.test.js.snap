// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[4] Matches snapshot from 5.8.22 1`] = `
<main
  className="all-pokemon"
>
  <article
    className="pokemon-card"
    data-testid="pokemon-card"
  >
    <h2
      className="poke-name card-item"
    >
      Fake Pokemon Name
       
      <span
        className="dex-id"
      >
        #
      </span>
    </h2>
    <h3
      className="poke-type card-item"
    >
      D
       / 
      e
       
    </h3>
    <h4
      className="height-weight card-item"
    >
      Height: 
      4m
       
      <br />
      Weight: 
      50kg
    </h4>
    <figure
      className="img-container card-item"
    >
      <img
        alt="Fake Pokemon Name"
        className="poke-img"
        data-testid="not-shiny"
        src="https://www.serebii.net/swordshield/pokemon/undefined.png"
      />
    </figure>
    <blockquote
      className="card-item"
    >
      blah blah blah
    </blockquote>
  </article>
  <article
    className="pokemon-card"
    data-testid="pokemon-card"
  >
    <h2
      className="poke-name card-item"
    >
      Fake2
       
      <span
        className="dex-id"
      >
        #
      </span>
    </h2>
    <h3
      className="poke-type card-item"
    >
      D
       / 
      e
       
    </h3>
    <h4
      className="height-weight card-item"
    >
      Height: 
      5m
       
      <br />
      Weight: 
      60kg
    </h4>
    <figure
      className="img-container card-item"
    >
      <img
        alt="Fake2"
        className="poke-img"
        data-testid="not-shiny"
        src="https://www.serebii.net/swordshield/pokemon/undefined.png"
      />
    </figure>
    <blockquote
      className="card-item"
    >
      This pokemon does nothing
    </blockquote>
  </article>
  <article
    className="pokemon-card"
    data-testid="pokemon-card"
  >
    <h2
      className="poke-name card-item"
    >
      Fake3
       
      <span
        className="dex-id"
      >
        #
      </span>
    </h2>
    <h3
      className="poke-type card-item"
    >
      G
       / 
      r
       
    </h3>
    <h4
      className="height-weight card-item"
    >
      Height: 
      6m
       
      <br />
      Weight: 
      80kg
    </h4>
    <figure
      className="img-container card-item"
    >
      <img
        alt="Fake3"
        className="poke-img"
        data-testid="not-shiny"
        src="https://www.serebii.net/swordshield/pokemon/undefined.png"
      />
    </figure>
    <blockquote
      className="card-item"
    >
      This pokemon is god
    </blockquote>
  </article>
</main>
`;
